src/shared/tpm2-util.c

```diff
@@ -1838,6 +1847,81 @@ static int find_signature(
 #endif
 }
 
+int tpm2_calculate_name(const TPMT_PUBLIC *public, TPM2B_NAME *ret_name) {
+        TSS2_RC rc;
+        int r;
+
+        assert(public);
+        assert(ret_name);
+
+        r = dlopen_tpm2();
+        if (r < 0)
+                return log_error_errno(r, "TPM2 support not installed: %m");
+
+        if (public->nameAlg != TPM2_ALG_SHA256)
+                return log_error_errno(SYNTHETIC_ERRNO(ENOTRECOVERABLE),
+                                       "Unsupported nameAlg: 0x%x",
+                                       public->nameAlg);
+
+        _cleanup_free_ uint8_t *buf = NULL;
+        size_t size = 0;
+
+        buf = malloc0(sizeof(*public));
```

Comment:
```c
buf = new(TPMT_PUBLIC, 1);
```

Reply:
and i did need to cast it, i.e. `buf = (uint8_t*) new0(TPMT_PUBLIC, 1)`